setup.py

```diff
@@ -1,8 +1,35 @@
 #!/usr/bin/env python
 import os
+import sys
 
 from setuptools import find_packages, setup
 
+min_version, max_version = ((3, 9), "3.9"), ((3, 11), "3.11")
+if not (min_version[0] <= sys.version_info[:2] <= max_version[0]):
+    # Python 3.5 does not support f-strings
+    py_version = ".".join([str(v) for v in sys.version_info[:3]])
+    error = (
+        "\n----------------------------------------\n"
+        "Error: This repo requires python {min_version}-{max_version}.\n"
+        "You are running python {py_version}".format(
+            min_version=min_version[1], max_version=max_version[1], py_version=py_version
+        )
+    )
+    print(error, file=sys.stderr)
+    sys.exit(1)
+# Update the README.rst
+with open("README.rst", "r") as f:
```

Comment:
Shouldn't this not be here? I assume this is the auto updating the readme feature?